src/main/java/org/purescript/parser/PureParsecParser.kt

```diff
@@ -584,9 +584,46 @@ class PureParsecParser {
             )
             .then(guardedDecl).`as`(ValueDeclaration)
     )
+    private val exportedClass =
+        token(CLASS)
+            .then(properName)
+            .`as`(PSElements.ExportedClass)
+    private val exportedData =
+        properName.`as`(PSElements.ExportedData)
+    private val exportedKind =
+        token(KIND)
+            .then(properName)
+            .`as`(PSElements.ExportedKind)
+    private val exportedModule =
+        token(MODULE)
+            .then(parseQualified(properName))
+            .`as`(PSElements.ExportedModule)
+    private val exportedOperator =
+        parens(operator)
+            .`as`(PSElements.ExportedOperator)
+    private val exportedType =
+        token(TYPE)
+            .then(parens(operator))
+            .`as`(PSElements.ExportedType)
+    private val exportedValue =
+        ident.`as`(PSElements.ExportedValue)
+    private val exportedItem =
+        choice(
+            exportedClass,
```

Comment:
I would rather have them inlined in the choice, then extracted, since they are quite simple and because that they are inline in the purescript parser

Reply:
Sure 👍